src/utils/event.ts

```diff
@@ -1,48 +1,81 @@
 import createKeccakHash from 'keccak';
+import { EventFragment, Result } from 'ethers/lib/utils';
+import { WarpInterface } from '../transcode/interface';
 import { toUintOrFelt } from './export';
 
-export type EventItem = { data: string[]; keys: string[]; order: number };
+export type EventItem = { data: string[]; keys: string[]; order?: number };
 
-export function decodeEventLog(eventsLog: EventItem[]): EventItem[] {
-  // flat number to hex string with rjust 62
-  const flatNumberToHexString = (num: number | bigint | string): string => {
+export function decodeWarpEvent(fragment: EventFragment | string, warpEvent: EventItem): Result {
+  warpEvent = join248bitChunks(warpEvent); // reverse 248 bit packing
+
+  // Remove leading 0x from each element and join them
+  const data = `0x${warpEvent.data.map((x) => x.slice(2)).join('')}`;
```

Comment:
We will need to do the same for keys right?

Reply:
No, `decodeEventLog` expects data as a string while topics as an array of strings